flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/source/ContinuousFileSplitEnumerator.java

```diff
@@ -91,6 +92,19 @@ private void addSplit(FileStoreSourceSplit split) {
                 .add(split);
     }
 
+    private void addSplitsBack(Collection<FileStoreSourceSplit> splits) {
+        List<FileStoreSourceSplit> backSplits = new ArrayList<>(splits);
+        Collections.reverse(backSplits);
+        backSplits.forEach(this::addSplitToHead);
+    }
+
+    private void addSplitToHead(FileStoreSourceSplit split) {
+        ((LinkedList<FileStoreSourceSplit>)
```

Comment:
Don't use cast, just declare `bucketSplits` to `Map<Integer, LinkedList<FileStoreSourceSplit>>`